bu-ssl.php

```diff
@@ -242,34 +293,106 @@ public function remove_all_postmeta() {
 	}
 
 	/**
-	 * Searches for insecure content in a post's content.
+	 * Searches for insecure content in a post's content
 	 *
 	 * @param string $content The post content.
-	 * @param string $type The type of content we are checking to see if it is insecure or not.
 	 * @return array The array of insecure URLs found.
 	 */
-	public function search_for_insecure_content( $content, $type = 'any' ) {
+	public function search_for_insecure_content( $content ) {
 		// Replace any internal http urls with relative urls.
 		$content = str_replace( get_site_url( null, null, 'http' ), get_site_url( null, null, 'relative' ), $content );
 
-		// Use regex to find all remaining insecure urls.
-		// This applies to external urls since we already took care of internal urls above.
-		preg_match_all( $this->options['http_all_regex'], $content, $urls, PREG_SET_ORDER );
+		// Load content into a DOMDocument object.
+		$dom = new \DOMDocument();
+		libxml_use_internal_errors( true );
+		$dom->loadHTML(
+			mb_convert_encoding( '<!DOCTYPE html><html lang="en"><body>' . $content . '</body></html>', 'HTML-ENTITIES', 'UTF-8' )
+		);
+		libxml_use_internal_errors( false );
 
-		// Filter list of insecure urls.
-		foreach ( $urls as $k => $u ) {
-			// If the url type does not match the specified type, remove it from the $urls array to ignore it.
-			if ( 'any' !== $type && $u[1] !== $type ) {
-				array_splice( $urls, $k, 1 );
-			}
+		// Declare the array which will hold all of the insecure urls.
+		$insecure_urls_per_tag = array();
+
+		// Iterate over the specified tags we are going to search for insecure urls.
+		foreach ( $this->tags as $tag ) {
+
+			// Query the content for the specified tag.
+			$dom_node_list = $dom->getElementsByTagName( $tag['name'] );
 
-			// If the url is internal, ignore it.
-			if ( 0 === strpos( $u[2], get_site_url( null, null, 'http' ) ) ) {
-				array_splice( $urls, $k, 1 );
+			// Initilize dom_node_list_array which will hold all of the DOMNodeList's we will check for insecure content.
+			$dom_node_list_array = array();
+
+			// If $tag['children'] is present.
+			if ( array_key_exists( 'children', $tag ) ) {
+				// Get children DOMNodeList's from the base DOMNodeList.
+				foreach ( $dom_node_list as $parent_element ) {
+					$dom_node_list_array[] = $parent_element->getElementsByTagName( $tag['children'] );
+				}
+			} else {
+				// Since there are no children, just add the base DOMNodeList to dom_node_list_array.
+				$dom_node_list_array[] = $dom_node_list;
 			}
+
+			// Iterate over all DOMNodeList's.
+			foreach ( $dom_node_list_array as $dom_node_list ) {
+				// Iterate over all elements in DOMNodeList.
+				foreach ( $dom_node_list as $element ) {
+					// Get the url from the specified attribute.
+					$attribute_value = $element->getAttribute( $tag['attribute'] );
+
+					// If attribute_value is not empty.
```

Comment:
Superfluous comment could be removed? Comments should be for code that needs explaining or to call attention to something important or odd, IMO.